File: ecole/006/ecole006.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n').map(Number);
const n = input[0];

const stack = [];
const result = [];
let num = 1;
let isPossible = true;

for (let i = 1; i <= n; i++) {
    const target = input[i];
    while (num <= target) {
        stack.push(num);
        result.push('+');
        num++;
    }

    if (stack[stack.length - 1] === target) {
        stack.pop();
        result.push('-');
    } else {
        isPossible = false;
        break;
    }
}

if (isPossible) {
    console.log(result.join('\n'));
} else {
    console.log('NO');
}

File: WEEK1/006/ecole006.py
n = int(input())
arr = [int(input()) for _ in range(n)]

stack = []
result = []
num = 1
possible = True

for target in arr:
    while num <= target:
        stack.append(num)
        result.append("+")
        num += 1

    if stack[-1] == target:
        stack.pop()
        result.append("-")
    else:
        possible = False
        break

if possible:
    for op in result:
        print(op)
else:
    print("NO")
